userspace/engine/formats.cpp

```diff
@@ -44,8 +44,10 @@ void falco_formats::init(sinsp *inspector,
 	s_json_include_output_property = json_include_output_property;
 	if(!s_formatters)
 	{
-		s_formatters = new sinsp_evt_formatter_cache(s_inspector);
+		delete(s_formatters);
+		s_formatters = NULL;
```

Comment:
Why assigning it to `NULL` after the `delete`?

There's the possibility of a double delete?

Also, maybe it's the case to think about a "smart" pointer here. Even to reason if we now need a pointer or not here...

Reply:
It's a WIP, and I was just trying the previous approach:
https://github.com/falcosecurity/falco/blob/0a449afc3ed0efc7d6ddf66fe206d4081ee405ae/userspace/engine/formats.cpp#L115-L123

Reply:
And ofc, now there's no need to assign it to `NULL` since two lines below it gets assigned to `new sinsp_evt_formatter_cache(s_inspector)` always (so no way to double delete it):

https://github.com/falcosecurity/falco/blob/c0f3b5acf4b53970549569e8feadbe5710c4d88c/userspace/engine/formats.cpp#L50

Alternatively, an `std::unique_ptr` should work here too.

However, you're probably right: we don't really need a pointer here. I will try that.

Thanks for the hints!

Reply:
Fixed.

N.B.
Although the pointer is not really needed, removing it implies having an empty `sinsp_evt_formatter_cache` for a while (when declaring the static member) and later freed at the first `init` call. That can lead to problems if `sinsp_evt_formatter_cache` implementation changed in the future. Thus I opted to keep the pointer (with `std::unique_ptr`).